### ISSUE TITLE
Contents missing in Stache after clearing the cache

### ISSUE THREAD

**Issue by efoken** [statamic/v2-hub#2196]
**Describe the bug**
We are continuously experiencing Stache issues...

**Sme important infos**
Our customers Website: https://ifworlddesignguide.com/ is running Statamic 2.10.5
We have an extra Server for site editors, and have the CP disabled for production. We use the Spock addon to commit and push new changes on the "edit" server. Then we have a shell script to push the `site/content` to the production server, clear the cache/Stache... Images/assets are coming from Minio (S3).

After clearing the cache **sometimes** there are contents missing. For example the big slider image on the homepage, or - sometimes - the whole header navigation, or sometimes a 404 error at index...

By the way: We use Redis for Caching @ production and we have written a script, with PHP Predis, fetching the Contents directly from Redis via the `laravel:stache::pages/data` key, to check if the contents are valid... but sometimes contents are there, sometimes not...

**Environment details (please complete the following information):**
 - Statamic Version 2.10.5
 - Fresh Install or Upgrade: Fresh
 - OS: Ubuntu 18.04
 - Browser: All
 - Web Server: Nginx
 - PHP Version: 7.2
 - Addons installed: SeoPro, Fetch, Placid, Spock

**Comment by jasonvarga** [comment]
> we have written a script, with PHP Predis, fetching the Contents directly from Redis via the laravel:stache::pages/data key

When is this run?

**Comment by efoken** [comment]
The script was just to check if contents are valid, so... clear cache, Open Browser, the Drache is rebuilt - contents are missing (sometimes). Sooo, to validate thus effect via CLI, I wrote a script to look directly to Redis. And **if so** we clear the Cache again and again, till it is working

**Comment by jasonvarga** [comment]
So you wrote that script to combat this same bug you're reporting?

**Comment by efoken** [comment]
Naaa, not exactly... the script was just to validate, if that's a bug in Statamic or something else with our templates or so... and because our customer wants an immediate fix, we have to clear the cache again and again, until everything goes right :-(

**Comment by jackmcdade** [comment]
Try upgrading to 2.10.7, I think you might find it's been fixed

**Comment by efoken** [comment]
No, that changed nothing. Contents are still randomly missing after clearing the cache...

**Comment by xahs** [comment]
I got the same problem upgrading from 2.10.4 to 2.10.7, asset containers disappeared from the cp. I have to recreate them manually but paths are saved as private (container::path) if I clear the cache again, containers are missing.

**Comment by efoken** [comment]
@xahs our problem is frontend related, in CP everything is fine

**Comment by jackmcdade** [comment]
Is this still occurring in 2.11.x?
